lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -146,6 +151,30 @@ def add_code_lens(node, name:, command:)
           type: "debug",
         )
       end
+
+      sig { params(node: SyntaxTree::Command).returns(T.nilable(String)) }
+      def resolve_gem_homepage(node)
+        gem_name = node.arguments.parts.flat_map(&:child_nodes).first.value
+        spec = Gem::Specification.stubs.find { |gem| gem.name == gem_name }&.to_spec
+        return if spec.nil?
+
+        spec.homepage || spec.metadata.fetch("homepage_uri", nil)
```

Comment:
The idea is to open the source control for the gem, not necessarily the homepage. So I think what we want is
```ruby
spec.homepage || spec.metadata.fetch("source_code_uri", nil)
```
I think it's also worth checking if the URL is from GitHub or GitLab and reject any others. The majority of gems are probably hosted on these two and we shouldn't allow people to click on random URLs. So something like
```ruby
[spec.homepage, spec.metadata["source_code_uri"]].find { |page| page&.start_with?("https://github.com") || page&.start_with?("https://gitlab.com") }
```